addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js

```diff
@@ -655,6 +671,83 @@ class EnvironmentScService extends Service {
     }
   }
 
+  async getCfnDetails(requestContext, environment, cfnStackLogicalId) {
+    const cfnClient = await this.getClientSdkWithEnvMgmtRole(
+      requestContext,
+      { id: environment.id },
+      { clientName: 'CloudFormation', options: { apiVersion: '2016-11-15' } },
+    );
+    const stackResources = await cfnClient.listStackResources({ StackName: cfnStackLogicalId }).promise();
+    const templateDetails = await cfnClient
+      .getTemplate({ StackName: cfnStackLogicalId, TemplateStage: 'Original' })
+      .promise();
+
+    return { stackResources, templateDetails };
+  }
+
+  async getWorkspaceSecurityGroup(requestContext, environment, securityGroupId) {
+    const ec2Client = await this.getClientSdkWithEnvMgmtRole(
+      requestContext,
+      { id: environment.id },
+      { clientName: 'EC2', options: { apiVersion: '2016-11-15' } },
+    );
+    const securityGroupResponse = await ec2Client.describeSecurityGroups({ GroupIds: [securityGroupId] }).promise();
+    return { securityGroupResponse };
+  }
+
+  async getSecurityGroupDetails(requestContext, environment) {
+    const outputsObject = cfnOutputsArrayToObject(environment.outputs);
+    const cfnStackLogicalId = outputsObject.CloudformationStackARN.split('/')[1];
+    const { stackResources, templateDetails } = await this.getCfnDetails(
+      requestContext,
+      environment,
+      cfnStackLogicalId,
+    );
+    const templateBody = YAML.load(templateDetails.TemplateBody);
+    const cfnTemplateIngressRules = templateBody.Resources.SecurityGroup.Properties.SecurityGroupIngress;
+
+    const securityGroup = _.find(
+      stackResources.StackResourceSummaries,
+      resource => resource.LogicalResourceId === 'SecurityGroup',
+    );
+    const securityGroupId = securityGroup.PhysicalResourceId;
+    const { securityGroupResponse } = await this.getWorkspaceSecurityGroup(
+      requestContext,
+      environment,
+      securityGroupId,
+    );
+
+    // Get protocol-port combinations from the SC CFN stack
+    const securityGroupDetails = securityGroupResponse.SecurityGroups[0];
+    const workspaceIngressRules = securityGroupDetails.IpPermissions;
+
+    // Only send back details of groups configured by the SC CFN stack
+    const returnVal = _.map(cfnTemplateIngressRules, cfnRule => {
+      const matchingRule = _.find(
+        workspaceIngressRules,
+        workspaceRule =>
+          cfnRule.FromPort === workspaceRule.FromPort &&
+          cfnRule.ToPort === workspaceRule.ToPort &&
+          cfnRule.IpProtocol === workspaceRule.IpProtocol,
+      );
+      const currentCidrRanges = matchingRule
+        ? _.concat(
+            _.map(matchingRule.IpRanges, ipRange => ipRange.CidrIp),
+            _.map(matchingRule.Ipv6Ranges, ipRange => ipRange.CidrIp),
```

Comment:
So for Ipv6Ranges what is the flow like? We display them on the UI but if the updates contain one, we don't allow those requests?

Reply:
Correct. User cannot continue if Ipv6 is included in any of the rules.